pkgs/by-name/ni/nix-required-mounts/package.nix

```diff
@@ -0,0 +1,57 @@
+{ addOpenGLRunpath
+, cmake
+, allowedPatterns ? rec {
+    opengl.onFeatures = [ "opengl" ];
+    opengl.paths = [
+      addOpenGLRunpath.driverLink
+      "/dev/video*"
+      "/dev/dri"
+    ];
+    cuda.onFeatures = [ "cuda" ];
+    cuda.paths = opengl.paths ++ [
+      "/dev/nvidia*"
+    ];
+  }
+, buildPackages
+, formats
+, lib
+, nix
+, python3
+, runCommand
+}:
+
+
+let
+  confPath = (formats.pythonVars { }).generate "config.py" {
+    CONFIG = {
+      inherit allowedPatterns;
+      nixExe = lib.getExe nix;
+    };
+  };
+  pname = "nix-required-mounts";
+in
+
+runCommand pname
+{
+  inherit confPath;
+  meta.mainProgram = pname;
+} ''
+  ${lib.getExe buildPackages.python3.pkgs.flake8} ${./main.py}
+
+  cat > main.py << EOF
+  #!${lib.getExe python3}
+
+  $(cat ${./main.py})
+  EOF
+
+  sed -ie '
+    /^entrypoint()$/ {
+      x ;
+      r ${confPath}
+      }' main.py
+
+  echo "entrypoint()" >> main.py
```

Comment:
Fairly ugly, but I didn't want any environment variables/runtime stuff, just a static "resource" included in the ~~binary~~ script. If you suggest a cleaner way I'll happily rewrite this

Reply:
This whole thing feels way more complicated than it needs to be... why not just use [makeWrapper](https://nixos.org/manual/nixpkgs/stable/#fun-makeWrapper) with environment variables?

Reply:
Yea, maybe (Someone just wants `std::embed`)